test/runtests.jl

```diff
@@ -3270,6 +3274,44 @@ else  # run HiGHS tests
+            inflated cashflows and it includes out-year, non-discounted battery replacement cost which is only included 
+            in the payback calulcation if the replacement happens before the payback period.
+            This scenario includes export benefits and CHP standby charges which are additive to the electricity bill for total electricity costs.
+            """
+
+            input_data = JSON.parsefile("./scenarios/after_tax_payback.json")
+            # First test with battery replacement within the payback period, but zero discount rate, so simple_payback_years should be equal to the X/Y payback metric
+            #  which discounts the future-year battery replacement back to present value so that it can be included in the payback calculation
+            s = Scenario(input_data)
+            inputs = REoptInputs(s)
+            m1 = Model(optimizer_with_attributes(HiGHS.Optimizer, "mip_rel_gap" => 0.01, "output_flag" => false, "log_to_console" => false))
+            m2 = Model(optimizer_with_attributes(HiGHS.Optimizer, "mip_rel_gap" => 0.01, "output_flag" => false, "log_to_console" => false))
+            results = run_reopt([m1,m2], inputs)
+            # Total operating (energy, fuel, O&M) cost savings output (available only with BAU scenario included)
+            savings = results["Financial"]["year_one_total_cost_savings_after_tax"]
+            # Net cost with non-discounted future capital-based incentives, including present value of battery replacement costs
+            capital_costs_after_non_discounted_incentives = results["Financial"]["capital_costs_after_non_discounted_incentives"]
+            # Calculated payback from above-two metrics
+            payback = capital_costs_after_non_discounted_incentives / savings
+            @test round(results["Financial"]["simple_payback_years"], digits=2) ≈ round(payback, digits=2)
+
+            # Test that with a non-zero discount rate, as long as the battery replacement cost is zero, these payback periods should also align
+            input_data["Financial"]["offtaker_discount_rate_fraction"] = 0.1
+            input_data["ElectricStorage"]["replace_cost_per_kw"] = 0.0
+            input_data["ElectricStorage"]["replace_cost_per_kwh"] = 0.0
+            s = Scenario(input_data)
+            inputs = REoptInputs(s)
+            m1 = Model(optimizer_with_attributes(HiGHS.Optimizer, "mip_rel_gap" => 0.01, "output_flag" => false, "log_to_console" => false))
+            m2 = Model(optimizer_with_attributes(HiGHS.Optimizer, "mip_rel_gap" => 0.01, "output_flag" => false, "log_to_console" => false))
+            results = run_reopt([m1,m2], inputs)
+            payback = results["Financial"]["capital_costs_after_non_discounted_incentives"] / results["Financial"]["year_one_total_cost_savings_after_tax"]
+            @test round(results["Financial"]["simple_payback_years"], digits=2) ≈ round(payback, digits=2)
```

Comment:
These metrics also differ in that `simple_payback_years` includes O&M costs, but the new "payback" calculation does not, right? This is a very light suggestion, but could consider adding a test with PV in which O&M costs are zeroed out to check that these are the same then? Maybe not necessary.

Reply:
It does include O&M, because year_one_total_operating_cost_savings_after_tax includes O&M costs. See results.jl line 124, scroll to the right.